### ISSUE TITLE
Exploratory testing: Cross-platform providers on .NET Core

### ISSUE THREAD

**Issue by divega** [dotnet/efcore#2896]
Currently we only own the System.Data.Common-based ADO.NET for SQLite and its corresponding EF provider. We should test (or keep testing) that it works across multiple platforms on .NET Core. As other providers become available we want to try them and help make sure things work great end-to-end.

**Comment by natemcmaster** [comment]
I believe this item was opened because SqlClient on Linux was so broken that exploratory testing was the only way to discover progress in its functionality. With a recent bug fix this week, I've re-enabled CI to run SQL Server *functional* tests on CoreCLR on Linux.

SQLite tests have been automated on Windows and Linux for some time now.

The only providers we haven't tested xplat are third-party providers.

**Comment by divega** [comment]
The idea was to also cover some exploratory testing for third party providers, especially those that we believe are important to have a credible cross platform story, but this is great progress!

**Comment by natemcmaster** [comment]
@divega this leaves just `EntityFramework7.Npgsql` for exploratory testing as SQL CE does not run on .NET Core.

http://docs.efproject.net/en/latest/providers/

**Comment by divega** [comment]
:+1:

**Comment by divega** [comment]
Until something else comes up :8ball:

**Comment by natemcmaster** [comment]
Blocked. Waiting on https://github.com/npgsql/npgsql/issues/937 and/or https://github.com/npgsql/npgsql/issues/938

**Comment by julielerman** [comment]
Nate, have your revisited the postgres provider since blocking this issue? Given that the package name (and dependencies) have changed and the namespaces changed, I haven't even tried adding it into project.json on a .net cli/rc2 project i'm fiddling with on OS X.

**Comment by natemcmaster** [comment]
@julielerman I have not revisited npgsql since blocking. We will revisit after dotnet-cli stabilizes more

**Comment by julielerman** [comment]
ok. I left an issue in their repo about updating those things. Back to Sqlite. Are you finding the same issues on OSX that are being reported on linux? (should I be asking elsewhere? :) )

**Comment by natemcmaster** [comment]
> Are you finding the same issues on OSX that are being reported on linux?

Can you clarify which issues? If something is crashing on SQLite w/ OSX, try looking in issues both on the [EF repo](https://github.com/aspnet/entityframework) and [Microsoft.Data.Sqlite](https://github.com/aspnet/microsoft.data.sqlite) and  to see if it has been reported/resolved. If not, you can open a new issue.
